src/adapters/implementations/paseto/token.service.ts

```diff
@@ -28,31 +29,49 @@ export class JWTAdapterService extends TokenAdapter {
 		super();
 	}
 
-	genAccess({
+	private getSecret(): KeyObject {
+		const secretString = this.config.get('PASETO_SECRET');
+		const secretBuffer = Buffer.from(secretString, 'base64');
+		const secretKeyObject = this.paseto.bytesToKeyObject(secretBuffer);
+		return secretKeyObject;
+	}
+
+	async genAccess({
 		accountId,
 		hasAcceptedLatestTerms,
-	}: GenAccessInput): GenAccessOutput {
+	}: GenAccessInput): Promise<GenAccessOutput> {
 		const payload: TokenPayload = {
 			sub: accountId,
 			terms: hasAcceptedLatestTerms,
 		};
 
+		const payloadRecord: Record<string, unknown> = {
```

Comment:
Duplicated, should use `payload`, if `paseto.sign` has any type errors, use `.sign(payload as Record<string, string>, secret)`